### PR TITLE
Cph search page + ViteJS

### PR DESCRIPTION
**Pull request by praveenmurugan101**
### Front-end
- migrates from react create-react-app to react ViteJS (as create-react-app is deprecated)
- creates a CPH page with autocomplete search box and react-tabulator for displaying CPH samples
- displays warnings for excluded samples
- fixes npm vulnerabilities

### Back-end
- creates a new request class to handle cph search
- creates a new endpoint to handle autocomplete cph search options
- creates a new endpoint to fetch all samples that contain the given cph and includes any warnings
- fixes flask vulnerabilities

### CODE REVIEW

File: app/front_end/src/components/CPHSearch/CPHSearch.jsx [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+                  dispatch(setShowPage("cattlemovement"));
+                },
+                nextstrain: async () => {
+                  dispatch(setNextstrainIdentifier(sample["Submission"]));
+                  dispatch(
+                    fetchNextstrainData({ identifier: sample["Submission"] })
+                  );
+                  dispatch(
+                    setNextstrainURL(
+                      `${sample["Clade"]}?f_Submission=${sample[
```
Comment (ellisrichardj) [inline app/front_end/src/components/CPHSearch/CPHSearch.jsx:92]: The URL could also be set to show all samples with the same CPH using
`${sample["Clade"]}?f_PreciseLocation=${cphValue["CPH"].replace(/ /g, "")}&p=grid`
I think this is something that the users would like to see

**Comment by televisedbrain** [comment]
Hi, this looks good; thanks for the updates. Nice work. In future can you please split up your PR more (in this case it would make sense to have a separate one for front & back end?) as it can be difficult to catch all the details in big PRs like this thanks

**Review by sainab-omar — APPROVED** [review]
Great work 💯

File: app/front_end/src/components/CPHSearch/CPHSearch.jsx (hunk 1 above)
Comment (praveenmurugan101) [reply app/front_end/src/components/CPHSearch/CPHSearch.jsx]: That's great suggestion, I have updated the code with these changes

**Pull request merged**